### PR TITLE
feat: Add deployment status modal with K8s conditions timeline

### PR DESCRIPTION
**Pull request by YuliaKrimerman**
https://issues.redhat.com/browse/RHOAIENG-15568

## Description

Adds a "Deployment status" modal that opens when clicking the status badge on model deployments. The modal displays K8s conditions as a vertical timeline using PatternFly's `ProgressStepper`, giving users visibility into each stage of a deployment's lifecycle.

The architecture is designed for extensibility: conditions are populated by each platform's status builder and flow through the generic `Deployment.status.conditions` field. Future Kueue integration can inject additional conditions without changes to the generic layer.

## How Has This Been Tested?

- Manually tested on a live cluster with KServe InferenceService deployments in Ready, Failed, and Starting states
- 

https://github.com/user-attachments/assets/3e2893a2-1e8a-4872-b444-7eda389cfec0

Screenshot 2026-07-09 at 12 02 10 PM
Screenshot 2026-07-09 at 12 02 20 PM
Screenshot 2026-07-13 at 5 08 55 PM

Screenshot 2026-07-09 at 12 08 07 PM
Screenshot 2026-07-09 at 12 14 05 PM

- Verified the modal opens on click from the deployments table, overview gallery cards, and Model Registry cards
- Verified condition timeline renders correctly with timestamps, error messages, and nested children
- Verified the "Stop deployment" and "Edit deployment" buttons work from the modal
- Verified the edit button shows a loading spinner while form data loads
- All unit tests pass across `model-serving` (248), `kserve` (38), and `llmd-serving` (124) packages

## Test Impact

**New tests added (26 total):**
All existing tests continue to pass unchanged.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)
- [x] The code follows our [Best Practices](/docs/best-practices.md) (React coding standards, PatternFly usage, performance considerations)

If you have UI changes:
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

## Summary by CodeRabbit

* **New Features**
  * Added deployment status details with progress steps, timestamps, warnings, errors, and nested readiness information.
  * Users can open deployment status details from status indicators across deployment views.
  * Added Stop and Edit actions where supported, including edit loading feedback.
  * Standardized deployment condition reporting across KServe and LLM-serving deployments.
* **Tests**
  * Added coverage for condition mapping, nested statuses, modal behavior, actions, and edge cases.

### CODE REVIEW

File: packages/llmd-serving/src/deployments/status.ts [hunk 1]
```diff
@@ -1,4 +1,15 @@
...
+  DeploymentConditionStatus,
+  DeploymentStatus,
+} from '@odh-dashboard/model-serving/extension-points';
+
+const toConditionStatus = (status?: string): DeploymentConditionStatus | undefined => {
+  if (status === 'True' || status === 'False') {
+    return status;
+  }
+  return status ? 'Unknown' : undefined;
+};
```
Comment (mturley) [inline packages/llmd-serving/src/deployments/status.ts]: You have this function defined in 2 places (here and in kserve), I think we could export it from the model-serving package. In both places it's also placed here above some imports which is odd

File: packages/model-serving/src/components/deployments/DeploymentStatusModal.tsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
+/* eslint-disable @odh-dashboard/no-restricted-imports -- custom modal with ProgressStepper timeline and embedded status badge in title; ContentModal does not support this layout */
```
Comment (mturley) [inline packages/model-serving/src/components/deployments/DeploymentStatusModal.tsx]: A file-wide eslint-disable is probably not desirable, can we put it on the necessary lines only?

**Review by mturley — CHANGES_REQUESTED** [review]
2 nits while I finish testing

**Review by mturley — APPROVED** [review]
I’m still figuring out how to test some corner cases like nested conditions, but I’ve tested thoroughly enough to be happy with this. I’ll follow up if necessary to open bugs

/approve
/lgtm
